content/guides/nextjs/github-open-authoring/github-oauth-app.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Set up the GitHub OAuth App
+---
+
+To get started, we need to set up an OAuth App in Github. Within your GitHub Account Settings, click <a href="https://github.com/settings/developers" target="_blank">OAuth Apps</a> under Developer Settings. Go ahead and create a "New OAuth App".
+
+Since you are **testing your app locally**, you'll create a _development_ GitHub app that redirects to localhost. Eventually you'll need to create separate OAuth Apps: one for development and a production app whose URLs will connect to the live domain. We'll circle back to the production app once when we cover [hosting](/guides/nextjs/github-open-authoring/hosting-vercel).
+
+For now, fill in `http://localhost:3000` for the _Homepage URL_. With the **Authorization callback URL**, enter `http://localhost:3000/github/authorizing`. This is the URL for an authorizing redirect page that we will create at a [later step](/guides/nextjs/github-open-authoring/auth-redirect).
+
+![oauth-app-config-example](/img/github-open-auth-cna/oAuth-app-config.png)
+
+Once the details are filled in, Go ahead and click 'Register Application'. After creating the app, you should see a page with information such as **Client ID** and **Client Secret**. Next, we'll add those as environment variables to the project to connect this App to the Tina-GitHub helpers.
+
+### Setting Environment Variables
+
+[Environment variables](https://nextjs.org/docs/api-reference/next.config.js/environment-variables) are sensitive values specific to your project. The Tina-GitHub helpers will use these values to talk to your repository, enabling auth and data fetching via GitHub.
+
+To set these variables, create a `.env` file in your project root. Add the _secret_ and _id_ values from the OAuth App, and fill in the repo name.
+
+> _Do not commit this file_; you may need to add `.env` to the `.gitignore` file.
+
+**.env**
+
+```
+# OAuth App Credentials from GitHub
+GITHUB_CLIENT_ID=
+GITHUB_CLIENT_SECRET=
+
+# The path to your repository in GitHub
+REPO_FULL_NAME=my-github-user/my-github-repo
```

Comment:
Do we still need to have `BASE_BRANCH` here?

Reply:
not sure, doesn't look like `BASE_BRANCH` was ever in this guide. seems to work without it.

Reply:
It does work without it, but IMO worth mentioning because BASE_BRANCH is useful if `master` isn't your base